internal/controller/scheduler/imagePullBackOffWatcher.go

```diff
@@ -147,3 +152,25 @@ func shouldCancel(containerStatus *v1.ContainerStatus) bool {
 	return containerStatus.State.Waiting != nil &&
 		containerStatus.State.Waiting.Reason == "ImagePullBackOff"
 }
+
+// All container-\d containers will have the agent installed as their PID 1.
+// Therefore, their lifecycle is well monitored in our backend, allowing us to terminate them if they fail to start.
+//
+// However, sidecar containers are completely unmonitored.
+// We avoid terminating jobs due to sidecar image pull backoff watcher
+// to prevent customer confusion.
+//
+// Most importantly, the CI can still pass (in theory) even if sidecars fail.
+//
+// (The name "system container" is subject to more debate.)
+func isSystemContainer(containerStatus *v1.ContainerStatus) bool {
+	name := containerStatus.Name
+	if name == AgentContainerName || name == CopyAgentContainerName {
+		return true
+	}
```

Comment:
should we include the checkout container here as well? we'll probably want to alert if we can't pull that image... something like:

```golang
	if slices.Contains([]slice{AgentContainerName, CopyAgentContainerName, CheckoutContainerName}, name) {
		return true
	}
```

Reply:
Good spot! Please take another look 🙏🏿 .